core/src/main/java/hudson/model/TimeZoneProperty.java

```diff
@@ -0,0 +1,118 @@
+package hudson.model;
+
+import hudson.Extension;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import hudson.util.ListBoxModel.Option;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.TimeZone;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+
+/**
+ * A UserProperty that allows a user to specify a time zone for displaying time.
+ */
+@Restricted(NoExternalUse.class)
+public class TimeZoneProperty extends UserProperty implements Saveable {
+    /**
+     * Time Zone ID defined by the user.
+     * {@code null} means that the time zone is not defined.
+     */
+    @CheckForNull
+    private String timeZoneName;
+
+    private static final Logger LOGGER = Logger.getLogger(TimeZoneProperty.class.getName());
+
+    @DataBoundConstructor
+    public TimeZoneProperty(@CheckForNull String timeZoneName) {
+        this.timeZoneName = timeZoneName;
+    }
+
+    private TimeZoneProperty() {
+        this(null);
+    }
+
+    @Override
+    public void save() throws IOException {
+        user.save();
+    }
+
+    public void setTimeZoneName(@CheckForNull String timeZoneName) {
+        this.timeZoneName = timeZoneName;
+    }
+
+    @Extension
+    @Symbol("timezone")
+    public static class DescriptorImpl extends UserPropertyDescriptor {
+
+        @Override
+        public String getDisplayName() {
+            return Messages.TimeZoneProperty_DisplayName();
+        }
+
+        @Override
+        public UserProperty newInstance(User user) {
+            return new TimeZoneProperty();
+        }
+
+        @Override
+        public synchronized void load() {
+            super.load();
+        }
+
+        @Override
+        public synchronized void save() {
+            super.save();
+        }
+
+        @Override
+        public boolean isEnabled() {
+            return true;
+        }
+
+        public ListBoxModel doFillTimeZoneNameItems() {
+            String current = forCurrentUser();
+            ListBoxModel items = new ListBoxModel();
+            items.add(Messages.TimeZoneProperty_DisplayDefaultTimeZone(), "");
+            for (String id : TimeZone.getAvailableIDs()) {
+                if (id.equalsIgnoreCase(current)) {
+                    items.add(new Option(id, id, true));
+                } else {
+                    items.add(id);
+                }
+            }
+            return items;
+        }
+    }
+
+    @Nullable
```

Comment:
```suggestion
    @CheckForNull
```

Reply:
@res0nance I would like to land it. Since it is a minot change, let's leave it for a follow-up PR by somebody